src/lib.rs

```diff
@@ -118,6 +146,63 @@ impl Drop for Model {
 	}
 }
 
+impl Drop for Metadata {
+	fn drop(&mut self) {
+		unsafe {
+			ds::DS_FreeMetadata(self.metadata);
+		}
+	}
+}
+
+impl Metadata {
+	pub fn items(&self) -> &[MetadataItem] {
+		unsafe {
+			let ptr   = (*self.metadata).items as * const MetadataItem;
+			slice::from_raw_parts(ptr, self.num_items() as usize)
+		}
+	}
+
+	pub fn num_items(&self) -> i32 {
+		unsafe {
+			(*self.metadata).num_items
+		}
+	}
+
+	pub fn probability(&self) -> f64 {
+		unsafe {
+			(*self.metadata).probability
+		}
+	}
+}
+
+impl MetadataItem {
+	pub fn character(&self) -> Result<&str, std::str::Utf8Error> {
+		unsafe {
+			let slice = CStr::from_ptr(self.metadata_item.character);
+			slice.to_str()
+		}
+	}
+
+	pub fn timestep(&self) -> i32 {
+		self.metadata_item.timestep
+	}
+
+	pub fn start_time(&self) -> f32 {
+		self.metadata_item.start_time
+	}
+}
+
+use std::fmt;
+impl fmt::Display for Metadata {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		let mut v = Vec::new();
+		for item in 0..self.num_items() {
+			v.extend_from_slice(self.items()[item as usize].character().unwrap().as_bytes());
```

Comment:
This isn't idiomatic Rust. You convert from a `str` back to a `&[u8]` and down below you convert back to a `String` again. Instead, make `v` a `String` and append the `str`'s to it via `+=`. Calling `write!` inside the loop would work as well, tbh I don't know which of the two is more idiomatic.